lib/jobs.js

```diff
@@ -61,9 +61,22 @@ function start(buildConfig, callback) {
  * @param {Function}   callback                   Callback fn(error, result)
  */
 function stop(buildConfig, callback) {
-    return redis.hdel(`${queuePrefix}buildConfigs`, buildConfig.buildId)
-        .then(() => executor.stop(buildConfig))
-        .then(result => callback(null, result), err => callback(err));
+    const stopId = buildConfig.buildId;
+    const stopConfig = { buildId: stopId };
+
+    return redis.hget(`${queuePrefix}buildConfigs`, stopId)
+        .then((fullBuildConfig) => {
+            const parsedConfig = JSON.parse(fullBuildConfig);
+
+            if (parsedConfig && parsedConfig.annotations) {
+                stopConfig.annotations = parsedConfig.annotations;
+            }
+        })
+        .catch(err => winston.error('err in stop job: failed to get build config', err))
+        .then(() => redis.hdel(`${queuePrefix}buildConfigs`, stopId))
+        .then(() => executor.stop(stopConfig))
+        .then(result => callback(null, result))
+        .catch(callback);
```

Comment:
I think this `catch` really should be part of the `then` above. This shouldn't be used to catch an error that occurs in the callback, for example, which would cause the callback to be called twice.

Reply:
Oh, I see.
I reverted this change.